File: r/R/is_array.R
is_array <- function(value, type = "any") {
  assert(is_string(type), "`type` parameter must be a string")

  if (type == "any") is.vector(value)
  else {
    mode <- switch(type,
      boolean = "logical",
      integer = "integer",
      number = "numeric",
      string = "character"
    )
    is.vector(value, mode = mode)
  }
}

File: tests/is_array.js
import test from 'tape'
import is_array from '../funcs/is_array.js'

test('is_array', function (t) {
  t.equal(is_array([]), true)
  t.equal(is_array([{}, []], "any"), true)
  t.equal(is_array([true, false], "boolean"), true)
  t.equal(is_array([1.2, 1.3], "number"), true)
  t.equal(is_array([1, 2, 3], "integer"), true)
  t.equal(is_array(['a', 'b', 'c'], "string"), true)
  t.end()
})
